src/test/change_stream.rs

```diff
@@ -0,0 +1,711 @@
+        .batch_size(10)
+        .build();
+    let mut change_stream = client.watch(pipeline, Some(options)).await.unwrap();
+    let result = RUNTIME
+        .timeout(Duration::from_millis(5), change_stream.next())
+        .await;
+    assert!(!result.is_ok());
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+#[function_name::named]
+async fn change_stream_insert_one() {
+    let _guard = LOCK.run_concurrently().await;
+
+    let ns = Namespace {
+        db: function_name!().to_string(),
+        coll: function_name!().to_string(),
+    };
+
+    let client = TestClient::new().await;
+    if client.server_version_lt(4, 0) || client.is_standalone() {
+        return;
+    }
+
+    let coll = client
+        .init_db_and_coll(function_name!(), function_name!())
+        .await;
+    client
+        .database(function_name!())
+        .create_collection(function_name!(), None)
+        .await
+        .unwrap();
+
+    let mut change_stream = coll.watch(None, None).await.unwrap();
+
+    let coll_ref = coll.clone();
+    RUNTIME.spawn(async move {
+        coll_ref
+            .insert_one(doc! { "_id": 1, "x": 1 }, None)
+            .await
+            .unwrap();
+    });
+
+    let document = change_stream.next().await.unwrap().unwrap();
+
+    assert_eq!(document.operation_type, OperationType::Insert);
+    assert_eq!(document.ns.unwrap(), ChangeStreamEventSource::Namespace(ns));
+    assert_eq!(document.to, None);
+    assert_eq!(document.document_key, Some(doc! { "_id": 1 }));
+    assert_eq!(document.update_description, None);
+    assert_eq!(document.full_document, Some(doc! { "_id": 1, "x": 1 }));
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+#[function_name::named]
+async fn change_stream_insert_many() {
+    let _guard = LOCK.run_concurrently().await;
+
+    let ns = Namespace {
+        db: function_name!().to_string(),
+        coll: function_name!().to_string(),
+    };
+
+    let client = TestClient::new().await;
+    if client.server_version_lt(4, 0) || client.is_standalone() {
+        return;
+    }
+
+    let coll = client
+        .init_db_and_coll(function_name!(), function_name!())
+        .await;
+
+    client
+        .database(function_name!())
+        .create_collection(function_name!(), None)
+        .await
+        .unwrap();
+
+    let mut change_stream = coll.watch(None, None).await.unwrap();
+
+    let coll_ref = coll.clone();
+    RUNTIME.spawn(async move {
+        let docs = vec![
+            doc! { "_id": 1, "title": "1984", "author": "George Orwell" },
+            doc! { "_id": 2, "title": "Animal Farm", "author": "George Orwell" },
+            doc! { "_id": 3, "title": "The Great Gatsby", "author": "F. Scott Fitzgerald" },
+        ];
+        coll_ref.insert_many(docs, None).await.unwrap();
+    });
+
+    for id in 1..4 {
```

Comment:
I think it would be a bit more clear to use an inclusive range here, i.e. `1..=3`

Reply:
fixed